File: frontend/types/language.ts
import { z } from "zod";

const Language = {
  CPP: "cpp",
  JAVA: "java",
  PYTHON: "python",
} as const;

export const LanguageEnum = z.enum([...Object.values(Language)]);
export type LanguageType = z.infer<typeof LanguageEnum>;
export default Language;

File: backend/src/main/java/com/shahrohit/hashcodex/enums/Language.java
package com.shahrohit.hashcodex.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum Language {
    CPP, JAVA, PYTHON;

    @JsonValue
    public String toJson() {
        return name().toLowerCase();
    }

    @JsonCreator
    public static Language fromJson(String value) {
        return Language.valueOf(value.toUpperCase());
    }
}
